store/sqlstore/plugin_store.go

```diff
@@ -111,6 +111,40 @@ func (ps SqlPluginStore) CompareAndSet(kv *model.PluginKeyValue, oldValue []byte
 	return true, nil
 }
 
+func (ps SqlPluginStore) CompareAndDelete(kv *model.PluginKeyValue, oldValue []byte) (bool, *model.AppError) {
+	if err := kv.IsValid(); err != nil {
+		return false, err
+	}
+
+	query := `DELETE FROM PluginKeyValueStore WHERE PluginId = :PluginId AND PKey = :Key AND PValue = :Old`
+	if oldValue == nil {
```

Comment:
do we care about this? I'm not sure it's a valid case for a plugin to store `nil`, since `KVGet` would return `nil` when it can't find a value.

As far I understand (which is very little :P) a plugin wouldn't be able to tell the difference between "there's no value stored for key `key1`" and "there's a value `nil` stored for `key1`"

Reply:
Agreed -- and now that I think about it, this was part of the impetus for making `KVCompareAndSet` delete on receiving a `nil` value. I do think having an explicit `KVCompareAndDelete` might still be a good idea.

Reply:
I see. I still think that having a separate function is best (easier to discover for consumers of the API) but I’m more than happy to go the original route if you prefer so.

Reply:
Thoughts about https://github.com/mattermost/mattermost-server/pull/11804/commits/d17f7c4d1cff47193dfa98e5e9988f58c4300a9c? With this change, when the caller uses `CompareAndSet` with `nil` as the new value, it's deleted.

Reply:
👍 for doing it on both routes.